### PR TITLE
Color UI component: reorder palettes and update names (core by defaults, user by custom)

### PR DESCRIPTION
**Pull request by oandregal**
Based on https://github.com/WordPress/gutenberg/pull/36496
Part of https://github.com/WordPress/gutenberg/issues/36543 and https://github.com/WordPress/gutenberg/issues/36556

In working with the new color UI component to show three different palettes we used these mockups https://github.com/WordPress/gutenberg/issues/34574 which use the following names:

- "default" to refer to the palette provided by WordPress core or the Gutenberg plugin
- "theme" to refer to the palette defined by the theme
- "custom" to refer to the palette created by the user

This PR updates:

- The labels of the UI components used in post and site editors, both "Core" to "Defaults" and "User" to "Custom".
- The `corePalette` and `coreGradients` flags in `theme.json` to be `defaultPalette` and `defaultGradients` instead.
- Some internal variable names.
- The order of the palettes: it should be theme, defaults, custom.

Follow-ups:

- The internal structure of the presets to use should be `default`, `theme`, and `user` instead of `core`, `theme`, and `user`. I'd like to address this as a follow-up to control the PR size. This matters because they're part of the API we expose in `useSetting( 'color.palette.core' )` so it should be coherent with the UI for simplicity.
- https://github.com/WordPress/gutenberg/issues/36641
- https://github.com/WordPress/gutenberg/pull/36639

### CODE REVIEW

File: packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js [hunk 1]
```diff
@@ -172,64 +172,70 @@ const PanelColorGradientSettingsSingleSelect = ( props ) => {
...
+	const customColors = useSetting( 'color.palette.user' );
 	const themeColors = useSetting( 'color.palette.theme' );
-	const coreColors = useSetting( 'color.palette.core' );
-	const shouldDisplayCoreColors = useSetting( 'color.corePalette' );
+	const defaultColors = useSetting( 'color.palette.core' );
+	const shouldDisplayDefaultColors = useSetting( 'color.defaultPalette' );
 
 	colorGradientSettings.colors = useMemo( () => {
 		const result = [];
-		if ( shouldDisplayCoreColors && coreColors && coreColors.length ) {
```
Comment (oandregal) [inline packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js:182]: Per the mockups, the order in which the palettes should be shown is: defaults, theme, user.

File: packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js [hunk 2]
```diff
@@ -172,64 +172,70 @@ const PanelColorGradientSettingsSingleSelect = ( props ) => {
...
 	const themeGradients = useSetting( 'color.gradients.theme' );
-	const coreGradients = useSetting( 'color.gradients.core' );
-	const shouldDisplayCoreGradients = useSetting( 'color.coreGradients' );
+	const defaultGradients = useSetting( 'color.gradients.core' );
+	const shouldDisplayDefaultGradients = useSetting(
+		'color.defaultGradients'
+	);
 	colorGradientSettings.gradients = useMemo( () => {
 		const result = [];
-		if (
```
Comment (oandregal) [inline packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js:209]: Per the mockups, the order in which the palettes should be shown is: defaults, theme, user.

File: packages/edit-site/src/components/global-styles/hooks.js [hunk 3]
```diff
@@ -218,57 +218,75 @@ export function getSupportedGlobalStylesPanels( name ) {
...
 
 export function useColorsPerOrigin( name ) {
-	const [ userColors ] = useSetting( 'color.palette.user', name );
+	const [ customColors ] = useSetting( 'color.palette.user', name );
 	const [ themeColors ] = useSetting( 'color.palette.theme', name );
-	const [ coreColors ] = useSetting( 'color.palette.core', name );
+	const [ defaultColors ] = useSetting( 'color.palette.core', name );
 	return useMemo( () => {
 		const result = [];
-		if ( coreColors && coreColors.length ) {
```
Comment (oandregal) [inline packages/edit-site/src/components/global-styles/hooks.js:226]: Per the mockups, the order of teh palettes is: theme, defaults, custom.

File: packages/edit-site/src/components/global-styles/hooks.js [hunk 4]
```diff
@@ -218,57 +218,75 @@ export function getSupportedGlobalStylesPanels( name ) {
...
 
 export function useGradientsPerOrigin( name ) {
-	const [ userGradients ] = useSetting( 'color.gradients.user', name );
+	const [ customGradients ] = useSetting( 'color.gradients.user', name );
 	const [ themeGradients ] = useSetting( 'color.gradients.theme', name );
-	const [ coreGradients ] = useSetting( 'color.gradients.core', name );
+	const [ defaultGradients ] = useSetting( 'color.gradients.core', name );
 	return useMemo( () => {
 		const result = [];
-		if ( coreGradients && coreGradients.length ) {
```
Comment (oandregal) [inline packages/edit-site/src/components/global-styles/hooks.js:254]: Per the mockups, the order of teh palettes is: theme, defaults, custom.

File: lib/class-wp-theme-json-gutenberg.php [hunk 5]
```diff
@@ -88,17 +88,17 @@ class WP_Theme_JSON_Gutenberg {
 			'width'  => null,
 		),
 		'color'      => array(
-			'background'     => null,
-			'corePalette'    => null,
-			'coreGradients'  => null,
```
Comment (Mamaduka) [inline lib/class-wp-theme-json-gutenberg.php:93]: I assume these settings were only recently introduced, which is why we don't need migrations.

**Review by Mamaduka — APPROVED** [review]
Thanks, @oandregal.

The changes look good to me, and the new order matches the mockups

**Review by carolinan — COMMENTED** [review]
I have tested this manually by viewing the palettes in the global styles sidebar, site editor, and in the block editor.
This works well, I am very glad the use of "Core" was replaced.

I then removed the two default palettes using the new names in theme.json.
The default palettes are removed from the site editor and block editor, but not from the global styles, is that correct?

--

I have one related question, that should not be a blocker for this PR. The border color option only displays the custom color palette, is that correct?

**Comment by oandregal** [comment]
@carolinan Thanks for the thorough testing, it's really helpful.

Apparently, we missed using those flags in the site editor, I'll get that fixed. Need to investigate the border-color issue.

**Comment by oandregal** [comment]
https://github.com/WordPress/gutenberg/pull/36639 fixes the flags in the GS sidebar.

**Comment by oandregal** [comment]
I've filled two issues for the color component:

https://github.com/WordPress/gutenberg/issues/36640
https://github.com/WordPress/gutenberg/issues/36641

**Pull request merged**